subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -403,6 +403,14 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 
 				/* No AD data overflow */
 				ad_len_overflow = 0U;
+
+				/* No chain PDU.
+				 * Note: Not required to assign as referencing
+				 * is guarded by the fact that ad_len_overflow
+				 * is zero; having the below to make compilers
+				 * not complain of uninitialized variable.
+				 */
+				ad_len_chain = 0U;
```

Comment:
Since `ad_len_chain` is now being set to `0` 7 places in this function, wouldn't it make more sense to initialize it to `0` where it is declared?

Reply:
Prefer to assign the value close to its locality of reference in each if-then-else clause, reduces redundant assignments.

Reply:
> reduces redundant assignments.

It's only redundant if you remember to set it to 0 all the other places ;) (would have avoided this issue in the first place).

Reply:
It is compile warning in older gcc 12, it was intentionally not assigned in all control path.